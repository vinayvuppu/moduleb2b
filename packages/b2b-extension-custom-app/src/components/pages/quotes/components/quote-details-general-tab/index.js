export { default } from './quote-details-general-tab';
