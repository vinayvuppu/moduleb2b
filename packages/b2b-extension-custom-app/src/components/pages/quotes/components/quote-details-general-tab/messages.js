import { defineMessages } from 'react-intl';

export default defineMessages({
  panelTitle: {
    id: 'Quote.Details.General.panelTitle',
    defaultMessage: 'Quotes summary',
  },
});
