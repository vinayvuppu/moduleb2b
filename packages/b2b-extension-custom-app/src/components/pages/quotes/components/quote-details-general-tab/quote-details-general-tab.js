import PropTypes from 'prop-types';
import React, { useState } from 'react';

import { useIsAuthorized } from '@commercetools-frontend/permissions';
import ReadOnlyMessage from '@commercetools-local/core/components/read-only-message';
import PageBottomSpacer from '@commercetools-local/core/components/page-bottom-spacer';
import TabContentLayout from '@commercetools-local/core/components/tab-content-layout';
import MetaDates from '@commercetools-local/core/components/meta-dates';
import { Spacings } from '@commercetools-frontend/ui-kit';
import { PERMISSIONS } from '../../../../../constants';
import QuoteSummaryPanel from '../quote-summary-panel';
import QuoteEmployeePanel from '../quote-employee-panel';
import QuoteItemsPanel from '../quote-items-panel';

export const TAB_NAMES = {
  GENERAL: 'general',
};

const NoMessage = () => <div />;
NoMessage.displayName = 'NoMessage';

const DEMANDED_PERMISSIONS = [
  PERMISSIONS.ManageCompanies,
  PERMISSIONS.ManageOrders,
];

export const QuoteDetailsGeneralTab = props => {
  const [tempTotalPrice, setTempTotalPrice] = useState();

  const isAuthorized = useIsAuthorized({
    demandedPermissions: DEMANDED_PERMISSIONS,
  });

  const handleUpdateTotal = total => {
    setTempTotalPrice(total);
  };

  return (
    <TabContentLayout
      header={
        <MetaDates
          created={props.quote.createdAt}
          modified={props.quote.lastModifiedAt}
        />
      }
      description={!isAuthorized ? <ReadOnlyMessage /> : <NoMessage />}
    >
      <Spacings.Stack scale="m">
        <QuoteSummaryPanel quote={props.quote} totalPrice={tempTotalPrice} />
        <QuoteEmployeePanel
          quote={props.quote}
          projectKey={props.projectKey}
          isAuthorized={isAuthorized}
        />
        <QuoteItemsPanel
          quote={props.quote}
          updateQuoteItems={props.updateQuoteItems}
          addAmountDiscount={props.addAmountDiscount}
          addPercentageDiscount={props.addPercentageDiscount}
          onUpdateOriginalTotalPrice={handleUpdateTotal}
          isAuthorized={isAuthorized}
        />
      </Spacings.Stack>
      <PageBottomSpacer />
    </TabContentLayout>
  );
};

QuoteDetailsGeneralTab.displayName = 'QuoteDetailsGeneralTab';
QuoteDetailsGeneralTab.propTypes = {
  quote: PropTypes.object,
  projectKey: PropTypes.string.isRequired,
  updateQuoteItems: PropTypes.func.isRequired,
  addAmountDiscount: PropTypes.func.isRequired,
  addPercentageDiscount: PropTypes.func.isRequired,
  hasCompany: PropTypes.bool.isRequired,
};

export default QuoteDetailsGeneralTab;
