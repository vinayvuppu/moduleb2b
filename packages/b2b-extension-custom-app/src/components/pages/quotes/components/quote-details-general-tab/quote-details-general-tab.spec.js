import React from 'react';
import { shallow } from 'enzyme';

import { QuoteDetailsGeneralTab } from './quote-details-general-tab';
import QuoteSummaryPanel from '../quote-summary-panel';
import QuoteEmployeePanel from '../quote-employee-panel';
import QuoteItemsPanel from '../quote-items-panel';

const createTestProps = props => ({
  projectKey: 'project-key-1',
  updateQuoteItems: jest.fn(),
  addAmountDiscount: jest.fn(),
  addPercentageDiscount: jest.fn(),
  quote: {
    id: 'quote-id',
  },
  hasCompany: true,
  ...props,
});

jest.mock('@commercetools-frontend/permissions', () => ({
  useIsAuthorized: () => true,
}));

describe('rendering', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<QuoteDetailsGeneralTab {...props} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render QuoteSummaryPanel', () => {
    expect(wrapper).toRender(QuoteSummaryPanel);
  });

  it('should render QuoteEmployeePanel', () => {
    expect(wrapper).toRender(QuoteEmployeePanel);
  });

  it('should render QuoteItemsPanel', () => {
    expect(wrapper).toRender(QuoteItemsPanel);
  });
});
