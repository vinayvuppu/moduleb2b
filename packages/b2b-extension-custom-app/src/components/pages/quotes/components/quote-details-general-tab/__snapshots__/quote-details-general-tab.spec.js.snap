// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should match snapshot 1`] = `
<TabContentLayout
  description={<NoMessage />}
  header={<branch(MetaDates) />}
>
  <Stack
    alignItems="stretch"
    scale="m"
  >
    <QuoteSummaryPanel
      quote={
        Object {
          "id": "quote-id",
        }
      }
    />
    <QuoteEmployeePanel
      isAuthorized={true}
      projectKey="project-key-1"
      quote={
        Object {
          "id": "quote-id",
        }
      }
    />
    <Connect(QuoteItemsPanel)
      addAmountDiscount={[MockFunction]}
      addPercentageDiscount={[MockFunction]}
      isAuthorized={true}
      onUpdateOriginalTotalPrice={[Function]}
      quote={
        Object {
          "id": "quote-id",
        }
      }
      updateQuoteItems={[MockFunction]}
    />
  </Stack>
  <PageBottomSpacer
    size="m"
  />
</TabContentLayout>
`;
